File: pset6/credit/credit.py
import math
from math import log10
from cs50 import get_int

creditcard = get_int("kaartnummer: ")
mem = creditcard
checksum = int(0)
length = math.floor(log10(creditcard))+1
firstdigits = int(0)
digit =int(0)

for i in range(length):
    digit = int(mem % 10)
    mem = int(mem / 10)
    if (i % 2 == 1):
        digit = 2 * digit
        if (digit > 9):
            checksum = int(checksum + int(digit % 10))
            checksum = int(checksum + int(digit / 10))
        else:
            checksum = int(checksum + digit)
        digit = digit / 2
    else:
        checksum = int(checksum + digit)

    if (i == length - 2):
        firstdigits = firstdigits + digit
    elif (i == length - 1):
        firstdigits = firstdigits + (digit * 10)

if (checksum % 10 != 0):
    print("INVALID")
elif (length == 15 and (firstdigits == 34 or firstdigits == 37)):
    print("AMEX")
elif (length == 13 and digit == 4):
    print("VISA")
elif (length == 16):
    if (firstdigits == 51 or firstdigits == 52 or firstdigits == 53 or firstdigits == 54 or firstdigits == 55):
        print("MASTERCARD")
    elif(digit == 4):
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    long creditcard = get_long("kaarnummer: ");
    long mem = creditcard;
    int digit, checksum = 0;
    int length = (int)log10((double)creditcard) + 1;
    int firstdigits = 0;

    
    for (int i = 0; i < length; i++)
    {
        digit = mem % 10;
        mem = mem / 10;
        if (i % 2 == 1)
        {
            digit = 2 * digit;
            if (digit > 9)
            {
                checksum = checksum + digit % 10;
                checksum = checksum + digit / 10;
            }
            else
            {
                checksum = checksum + digit;                
            }
            digit = digit / 2;
        }
        else
        {
            checksum = checksum + digit;
        }
        
        
        if (i == length - 2)
        {
            firstdigits = firstdigits + digit;
        }
        else if (i == length - 1)
        {
            firstdigits = firstdigits + digit * 10;
        }
        
    }
    
    
    
    
    if (checksum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else if (length == 15 && (firstdigits == 34 || firstdigits == 37))
    {
        printf("AMEX\n");
    }
    else if (length == 13 && digit == 4)
    {
        printf("VISA\n");
    }
    else if (length == 16)
    {
        if (firstdigits == 51 || firstdigits == 52 || firstdigits == 53 || firstdigits == 54 || firstdigits == 55)
        {
            printf("MASTERCARD\n");
        }
        else if (digit == 4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
    
    
    
    
}